pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -383,11 +383,20 @@ stdenv.mkDerivation ({
       mv tmp "$d"
     done
 
-    for d in $(grep '^dynamic-library-dirs:' "$packageConfDir"/* | cut -d' ' -f2- | tr ' ' '\n' | sort -u); do
-      for lib in "$d/"*.{dylib,so}; do
-        ln -s "$lib" "$dynamicLinksDir"
+    listDynamicLinks () {
+      for d in $(grep '^dynamic-library-dirs:' "$packageConfDir"/* | cut -d' ' -f2- | tr ' ' '\n' | sort -u); do
+        for lib in "$d/"*.{dylib,so} ; do
+          if [ -f "$lib" ] ; then
+            echo "$lib"
+          fi
+        done
       done
+    }
+
+    for lib in $(listDynamicLinks | sort | uniq) ; do
+      ln -s "$lib" "$dynamicLinksDir"
     done
```

Comment:
I think the `ln -s` step can best be pushed inside the function. Then this is just a call, like:
```shell
    makeDynamicLinks
```
That just keeps the `ln -s` a little closer to the logic that drives it.

(Also, as is I think the `sort | uniq` shouldn't be needed, because of the `sort -u` that's inside the function.)

Reply:
`sort | uniq` deletes duplicated links.
`sort -u` deletes duplicated directories.
If we delete either, we should delete `sort -u`.

Reply:
Thank you for your reviewing.
I will update it and check the test.